catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/FederationAdmin.java

```diff
@@ -468,9 +471,10 @@ private Metacard getRegistryMetacardFromInputStream(InputStream inputStream)
         Map<String, Metacard> registryIdMetacardMap = new HashMap<>();
 
         for (Metacard metacard : metacards) {
-            String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                    .getValue()
-                    .toString();
+            if (!RegistryUtility.hasRegistryId(metacard)) {
+                continue;
+            }
+            String registryId = RegistryUtility.getRegistryId(metacard);
```

Comment:
You're testing the registry id attribute for null twice, in hasRegistryId & getRegistryId. Can you remove the hasRegistryID check and just call RegistryUtility.getRegistryId and check the return for null?